src/panels/config/labs/ha-config-labs.ts

```diff
@@ -38,8 +41,12 @@ class HaConfigLabs extends SubscribeMixin(LitElement) {
   public hassSubscribe() {
     return [
       subscribeLabFeatures(this.hass.connection, (features) => {
-        // Sort by integration domain alphabetically
-        this._preview_features = features.sort((a, b) => a.domain.localeCompare(b.domain));
+        // Sort by localized integration name alphabetically
+        this._preview_features = features.sort((a, b) =>
+          domainToName(this.hass.localize, a.domain).localeCompare(
+            domainToName(this.hass.localize, b.domain)
```

Comment:
Ideally this would be done in a memoized function that takes `localize` and `_preview_features` as arguments, so that when the translations change the items will be sorted properly again.